### PR TITLE
Add symbolic shape comparison optimization

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64323 Add embedding shape analysis
* #64121 Max Pool with indices
* #63530 Add Maxpool to shape analysis / Opinfo
* **#64300 Add symbolic shape comparison optimization**
* #64299 Refactor to use shape arguments
* #63516 Add view with negative dim
* #63615 Generalize expand logic
* #63407 Add permute, arange
* #63365 Add support for slice, selec twith int, index_select
* #63099 Add squeeze, unsqueeze, transpose shape functins
* #63050 Add batch of unary functions

Differential Revision: [D30738146](https://our.internmc.facebook.com/intern/diff/D30738146)

### CODE REVIEW

File: torch/csrc/jit/passes/symbolic_shape_analysis.cpp [hunk 1]
```diff
@@ -369,6 +367,36 @@ struct SymbolicShapeAnalyzer {
               int64_t symbolic_index = shape_symbol.value();
               symbolic_shape_map[symbolic_index].push_back(use.user->output());
             }
+            for (const auto& sym_uses: use.user->output()->uses()) {
+              auto k = sym_uses.user->kind();
+              if (k != aten::ge && k != aten::le && k != aten::ne && k != aten::eq) {
```
Comment (Krovatkin) [inline torch/csrc/jit/passes/symbolic_shape_analysis.cpp:372]: why can't we do `gt` and `lt`?

**Review by Krovatkin — APPROVED** [review]
:shipit:

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738146).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738146).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30738146).

**Pull request closed without merging**